src/zeroconf/_protocol/incoming.py

```diff
@@ -142,24 +144,43 @@ def is_response(self) -> bool:
 
     def has_qu_question(self) -> bool:
         """Returns true if any question is a QU question."""
-        if not self.num_questions:
-            return False
-        for question in self.questions:
-            # QU questions use the same bit as unique
-            if question.unique:
-                return True
-        return False
+        return self._has_qu_question
 
     @property
     def truncated(self) -> bool:
         """Returns true if this is a truncated."""
         return (self.flags & _FLAGS_TC) == _FLAGS_TC
 
+    @property
+    def questions(self) -> List[DNSQuestion]:
+        """Questions in the packet."""
+        return self._questions
+
+    @property
+    def num_questions(self) -> int:
+        """Number of questions in the packet."""
+        return self._num_questions
+
+    @property
+    def num_answers(self) -> int:
+        """Number of answers in the packet."""
+        return self._num_answers
+
+    @property
+    def num_authorities(self) -> int:
+        """Number of authorities in the packet."""
+        return self._num_authorities
+
+    @property
+    def num_additionals(self) -> int:
+        """Number of additionals in the packet."""
+        return self._num_additionals
```

Comment:
These shouldn't be accessed publicly but do in tests so I was worried about removing them as it could be a breaking change